upgrades.js

```diff
@@ -0,0 +1,82 @@
+module.exports = {
+
+	/**
+	 * INTERNAL: add various upgrade scripts
+	 *
+	 * @access protected
+	 */
+	addUpgradeScripts() {
+
+		if (process.env.DEVELOPER) {
+			self.config._configIdx = -1;
+		}
```

Comment:
I think this needs to be in the main file. This makes the update scripts to run every time only when running in `dev` mode, but `self` doesn't exist here.

Reply:
Yeah, missed that.  I do a merge of this file into the class, so it should otherwise run with the correct context and is called in the constructor just the same.